ipatests/test_integration/test_backup_and_restore.py

```diff
@@ -446,36 +449,124 @@ def test_full_backup_reinstall_restore_with_vault(self):
+            - check new replica  setup after restore
+        https://pagure.io/freeipa/issue/7455
+            check that after restore and replication reinitialization
+                - users and CA data are at state before backup
+                - CA can be installed on existing replica
+                - new replica with CA can be setup
+    """
+    num_replicas = 2
     topology = "star"
+    csr = "my.csr"
 
     @classmethod
     def install(cls, mh):
+        cls.replica1 = cls.replicas[0]
+        cls.replica2 = cls.replicas[1]
         if cls.domain_level is None:
             domain_level = cls.master.config.domain_level
         else:
             domain_level = cls.domain_level
 
-        if cls.topology is None:
-            return
-        else:
-            tasks.install_topo(
-                cls.topology, cls.master, [],
-                cls.clients, domain_level
-            )
+        # Configure only master and one replica.
+        # Replica is configured without CA
+        tasks.install_topo(
+            cls.topology, cls.master, [cls.replica1],
+            cls.clients, domain_level,
+            setup_replica_cas=False
+        )
+
+    def prepare_cert_signing_request(self):
+        self.master.run_command(
+            ['ipa', 'service-add', 'TEST/' + self.master.hostname])
+        openssl_command = ['openssl', 'req', '-new', '-nodes',
+                           '-out', self.csr,
+                           '-subj', '/CN=' + self.master.hostname]
+        self.master.run_command(openssl_command)
+        self.replica1.run_command(openssl_command)
+        self.replica2.run_command(openssl_command)
+
+    @staticmethod
+    def get_users(host):
+        res = host.run_command(['ipa', 'user-find'])
+        users = set()
+        for line in res.stdout_text.splitlines():
+            k, _, v = line.strip().partition(': ')
+            if k == 'User login':
+                users.add(v)
+        return users
+
+    @staticmethod
+    def wait_replication_status(host, target_status):
+        timeout = 30
+        status_attr = 'nsds5replicaLastUpdateStatus'
+        ldap_filter = '(objectclass=nsds5replicationagreement)'
+
+        ldap = host.ldap_connect()
+        start = time.time()
+        while True:
+            entries = ldap.get_entries(
+                DN(('cn', 'mapping tree'), ('cn', 'config')),
+                filter=ldap_filter,
+                attrs_list=[status_attr])
+            statuses = [entry.single_value[status_attr] for entry in entries]
+            wrong_statuses = [s for s in statuses if target_status not in s]
+            if wrong_statuses:
+                if time.time() - start > timeout:
+                    raise AssertionError(
+                        'Unexpected status after %s seconds: %s' %
+                        (timeout, wrong_statuses[0]))
+                time.sleep(1)
+                continue
+            break
+
+    @classmethod
```

Comment:
``classmethod`` should only be used for alternative constructors or for other special methods like class setup / tear down. A normal method is just fine.